docs/development/plugin/development-elasticsearch.asciidoc

```diff
@@ -0,0 +1,41 @@
+[[development-elasticsearch]]
+=== Communicating with elasticsearch
+
+Kibana exposes two clients on the server and browser for communicating with elasticsearch.
+There is an 'admin' client which is used for managing Kibana's state, and a 'data' client for all
+other requests.  The clients use the elasticsearch.js library.
```

Comment:
the intent for this line was to indicate which API's need to be used when using these clients.  i moved this entire page to the core development section, and add a link to the elasticearch.js docs.   thoughts on if this needs more?